lib/Mojolicious.pm

```diff
@@ -0,0 +1,1217 @@
+L</"dispatch"> the last, yours will be in between.
+
+  $app->hook(around_dispatch => sub ($next, $c) {
+    ...
+    $next->();
+    ...
+  });
+
+This is a very powerful hook and should not be used lightly, it allows you to, for example, customize application-wide
+exception handling, consider it the sledgehammer in your toolbox. (Passed a callback leading to the next hook and the
+default controller object)
+
+=head2 before_dispatch
+
+Emitted right before the static file server and router start their work.
+
+  $app->hook(before_dispatch => sub ($c) {...});
+
+Very useful for rewriting incoming requests and other preprocessing tasks. (Passed the default controller object)
+
+=head2 after_static
+
+Emitted after a static file response has been generated by the static file server.
+
+  $app->hook(after_static => sub ($c) {...});
+
+Mostly used for post-processing static file responses. (Passed the default controller object)
+
+=head2 before_routes
+
+Emitted after the static file server determined if a static file should be served and before the router starts its
+work.
+
+  $app->hook(before_routes => sub ($c) {...});
+
+Mostly used for custom dispatchers and collecting metrics. (Passed the default controller object)
+
+=head2 around_action
+
+Emitted right before an action gets executed and wraps around it, so you have to manually forward to the next hook if
+you want to continue the chain. Default action dispatching is the last hook in the chain, yours will run before it.
+
+  $app->hook(around_action => sub ($next, $c, $action, $last) {
+    ...
+    return $next->();
+  });
+
+This is a very powerful hook and should not be used lightly, it allows you for example to pass additional arguments to
+actions or handle return values differently. Note that this hook can trigger more than once for the same request if
+there are nested routes. (Passed a callback leading to the next hook, the current controller object, the action
+callback and a flag indicating if this action is an endpoint)
+
+=head2 before_render
+
+Emitted before content is generated by the renderer. Note that this hook can trigger out of order due to its dynamic
+nature, and with embedded applications will only work for the application that is rendering.
+
+  $app->hook(before_render => sub ($c, $args) {...});
+
+Mostly used for pre-processing arguments passed to the renderer. (Passed the current controller object and the render
+arguments)
+
+=head2 after_render
+
+Emitted after content has been generated by the renderer that will be assigned to the response. Note that this hook can
+trigger out of order due to its dynamic nature, and with embedded applications will only work for the application that
+is rendering.
+
+  $app->hook(after_render => sub ($c, $output, $format) {...});
+
+Mostly used for post-processing dynamically generated content. (Passed the current controller object, a reference to
```

Comment:
Perhaps in the cookbook document an example of post-processing dynamically generated content.  Also, it seems it could be helpful to the reader to end with (i.e. post-processing in this hook is not applicable to static content).